### PR TITLE
Allow rxn rate params/exprs with sympy fn names

### PR DESCRIPTION
**Pull request by alubbock**
Defining a parameter or expression using a sympy function name (e.g.
`deg`) will cause the following error:

    SympifyError: SympifyError: <function deg at 0x10f0d0140>

The error is down to `_parse_reaction` in bng.py, which runs `sympy.S(x)`
where `x` is a string such as `'deg'`. Converting the string to the relevant
parameter or expression object resolves this.

Resolves: pysb/pysb#259

### CODE REVIEW

File: pysb/simulator/cupsoda.py [hunk 1]
```diff
@@ -474,12 +474,13 @@ def _get_cmatrix(self):
             for j in range(self._len_rxns):
                 rate = 1.0
                 for r in rate_args[j]:
-                    x = str(r)
-                    if x in par_names:
-                        rate *= par_vals[i][par_dict[x]]
+                    if isinstance(r, pysb.Parameter):
+                        rate *= par_vals[i][par_names.index(r.name)]
```
Comment (jmuhlich) [inline pysb/simulator/cupsoda.py]: Was going through `par_dict` causing a problem, or did you decide it was just an unnecessary optimization?

Comment (alubbock) [inline pysb/simulator/cupsoda.py]: It didn't cause a problem, but I was simplifying the code when making some other changes and didn't revert this change back. I would imagine the performance difference is negligible in any case.
